tests/lca/imagebasedupgrade/cnf/internal/cnfclusterinfo/cnfclusterinfo.go

```diff
@@ -75,11 +104,87 @@ func (upgradeVar *ClusterStruct) SaveClusterInfo() error {
 		return errors.New("node list is empty")
 	}
 
+	sriovNet, err := sriov.List(cnfinittools.TargetSNOAPIClient, cnfinittools.CNFConfig.SriovOperatorNamespace)
+	if err != nil {
+		return err
+	}
+
+	for _, net := range sriovNet {
+		upgradeVar.SriovNetworks = append(upgradeVar.SriovNetworks, net.Object.Name)
+	}
+
+	sriovPolicy, err := sriov.ListPolicy(cnfinittools.TargetSNOAPIClient, cnfinittools.CNFConfig.SriovOperatorNamespace)
+
+	for _, policy := range sriovPolicy {
+		upgradeVar.SriovNetworkNodePolicies = append(upgradeVar.SriovNetworkNodePolicies, policy.Object.Name)
+	}
+
 	upgradeVar.Version = clusterVersion.Object.Status.Desired.Version
 	upgradeVar.ID = string(clusterVersion.Object.Spec.ClusterID)
 	upgradeVar.Name = targetSnoClusterName
 	upgradeVar.Operators = installedCSV
 	upgradeVar.NodeName = node[0].Object.Name
 
+	_ = upgradeVar.getWorkloadInfo()
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (upgradeVar *ClusterStruct) getWorkloadInfo() error {
+	for _, workloadNS := range strings.Split(cnfinittools.CNFConfig.IbuWorkloadNS, ",") {
+		workloadMap := WorkloadStruct{
+			Namespace: workloadNS,
+			Objects: WorkloadObjects{
+				Deployment:  []string{},
+				StatefulSet: []string{},
+			},
+		}
+
+		deployments, err := deployment.List(cnfinittools.TargetSNOAPIClient, workloadNS)
+		if err != nil {
+			return err
+		}
+
+		for _, deployment := range deployments {
+			workloadMap.Objects.Deployment = append(workloadMap.Objects.Deployment, deployment.Definition.Name)
+		}
+
+		statefulsets, err := statefulset.List(cnfinittools.TargetSNOAPIClient, workloadNS)
+		if err != nil {
+			return err
+		}
+
+		for _, statefulset := range statefulsets {
+			workloadMap.Objects.StatefulSet = append(workloadMap.Objects.StatefulSet, statefulset.Definition.Name)
+		}
+
+		upgradeVar.WorkloadResources = append(upgradeVar.WorkloadResources, workloadMap)
+	}
+
+	pod, err := pod.Pull(
```

Comment:
Just to be safe maybe call this workloadPod ?

Reply:
ah, good point. Updated, tnx!